src/Bicep.Cli/Program.cs

```diff
@@ -99,6 +101,11 @@ public async Task<int> RunAsync(string[] args)
                 invocationContext.ErrorWriter.WriteLine(exception.Message);
                 return 1;
             }
+            catch (ConfigurationException exception)
```

Comment:
Worth establishing a base class to represent 'exception with preformatted message that can be shown to the end user'? Feels like otherwise we're just proliferating catch blocks with no logical differences.

Reply:
Moved the `BicepException` class from `Bicep.Cli` to `Bicep.Core` and made it a base exception class.